drivers/power_domain/power_domain_intel_adsp.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/pm/device.h>
+#include <zephyr/pm/device_runtime.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(power_domain_intel_adsp, LOG_LEVEL_INF);
+
+struct pg_registers {
+	uint32_t wr_address;
+	uint32_t rd_address;
+	uint32_t SPA_bit;
+	uint32_t CPA_bit;
+};
+
+static int pd_intel_adsp_set_power_enable(struct pg_registers *reg, bool power_enable)
+{
+	uint32_t SPA_bit_mask = BIT(reg->SPA_bit);
+
+	if (power_enable) {
+		sys_write32(sys_read32(reg->wr_address) | SPA_bit_mask, reg->wr_address);
+	} else {
+		sys_write32(sys_read32(reg->wr_address) & ~(SPA_bit_mask), reg->wr_address);
+	}
+
+	return 0;
+}
+
+static int pd_intel_adsp_pm_action(const struct device *dev, enum pm_device_action action)
+{
+	struct pg_registers *reg_data = (struct pg_registers *)dev->data;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		pm_device_children_action_run(dev, PM_DEVICE_ACTION_TURN_ON, NULL);
+		pd_intel_adsp_set_power_enable(reg_data, true);
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		pm_device_children_action_run(dev, PM_DEVICE_ACTION_TURN_OFF, NULL);
+		pd_intel_adsp_set_power_enable(reg_data, false);
+		break;
+	case PM_DEVICE_ACTION_TURN_ON:
+		break;
+	case PM_DEVICE_ACTION_TURN_OFF:
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+static int pd_intel_adsp_init(const struct device *dev)
+{
+	pm_device_init_suspended(dev);
+	pm_device_runtime_enable(dev);
+	return 0;
+}
+
+#define DT_DRV_COMPAT intel_adsp_power_domain
```

Comment:
```
#define DT_DRV_COMPAT intel_adsp_power_domain
```
	Where this device is being declared ? I'm not seeing any DT change in this PR.

Reply:
My focus there is on ACE platform, which is being upstreamed right now. DT will come right after the board definition is there to work on. This file is not compiled for cavs boards.

Reply:
Understood. Note that the binding is should be part of this PR since it is introducing the new driver.

I got it that this is not being used anywhere now and we need the board definition to make use of this driver, but the binding can/should be part of this PR IMHO.